modules/video/src/lkpyramid.cpp

```diff
@@ -239,13 +239,12 @@ void cv::detail::LKTrackerInvoker::operator()(const Range& range) const
         acctype iA11 = 0, iA12 = 0, iA22 = 0;
         float A11, A12, A22;
 
-#if CV_SSE2
-        __m128i qw0 = _mm_set1_epi32(iw00 + (iw01 << 16));
-        __m128i qw1 = _mm_set1_epi32(iw10 + (iw11 << 16));
-        __m128i z = _mm_setzero_si128();
-        __m128i qdelta_d = _mm_set1_epi32(1 << (W_BITS1-1));
-        __m128i qdelta = _mm_set1_epi32(1 << (W_BITS1-5-1));
-        __m128 qA11 = _mm_setzero_ps(), qA12 = _mm_setzero_ps(), qA22 = _mm_setzero_ps();
+#if CV_SSE2 || CV_VSX
+        v_int32x4 qw0 = v_setall_s32(iw00 + (iw01 << 16));
+        v_int32x4 qw1 = v_setall_s32(iw10 + (iw11 << 16));
```

Comment:
Looks like this part of code assumes Little-endian platforms only.

And there are a lot of `v_reinterpret_as_s16(qw0)` statements below.

Reply:
Changing to v_int16x8 with interleaved data.